### ISSUE TITLE
Starting karaf on a 32bit ubuntu linux fails

### ISSUE THREAD

**Issue by cschneider** [jline/jline3#108]
...temp/apache-karaf-4.1.1/bin$ ./karaf 
        __ __                  ____      
       / //_/____ __________ _/ __/      
      / ,<  / __ `/ ___/ __ `/ /_        
     / /| |/ /_/ / /  / /_/ / __/        
    /_/ |_|\__,_/_/   \__,_/_/         

  Apache Karaf (4.1.1)

Hit '<tab>' for a list of available commands
and '[cmd] --help' for help on a specific command.
Hit '<ctrl-d>' or type 'system:shutdown' or 'logout' to shutdown Karaf.

Error executing command: [14] Bad address

In the log there is this exception:

org.apache.karaf.shell.core - 4.1.1 | Exception caught while executing command
com.sun.jna.LastErrorException: [14] Bad address
	at com.sun.jna.Native.invokeVoid(Native Method) ~[?:?]
	at com.sun.jna.Function.invoke(Function.java:407) [?:?]
	at com.sun.jna.Function.invoke(Function.java:353) [?:?]
	at com.sun.jna.Library$Handler.invoke(Library.java:244) [?:?]
	at com.sun.proxy.$Proxy35.ioctl(Unknown Source) [49:org.jline:3.2.0]
	at org.jline.terminal.impl.jna.linux.LinuxNativePty.getSize(LinuxNativePty.java:82) [49:org.jline:3.2.0]
	at org.jline.terminal.impl.AbstractPosixTerminal.getSize(AbstractPosixTerminal.java:59) [49:org.jline:3.2.0]
	at org.apache.karaf.shell.impl.console.JLineTerminal.getSize(JLineTerminal.java:167) [42:org.apache.karaf.shell.core:4.1.1]
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:474) [49:org.jline:3.2.0]
	at org.apache.karaf.shell.impl.console.ConsoleSessionImpl.run(ConsoleSessionImpl.java:339) [42:org.apache.karaf.shell.core:4.1.1]

**Comment by andreasmattes** [comment]
I've looked into this issue and found the following problem in class "org.jline.terminal.impl.jna.linux.CLibrary": The 2nd argument to function "ioctl(...)" must be int, not long. It is 64-bit wide under Linux 64-bit, but 32-bit wide under Linux 32-bit resp. a 32-bit JVM. So pushing the long onto the C stack under 32-bit causes the subsequent pointer to be shifted by 32-bit against where it is expected by the "ioctl(...)" C function, is read out 32-bit left-shifted, and causes the segmentation fault error (EFAULT) returned as "bad address".

I've tried out locally to change the second argument of "ioctl(...)" and the corresponding constants to int, and it has worked with the 32-bit and the 64-bit JVM under Linux (openSUSE Leap 42.2, 64-bit).
